build_projects/dotnet-host-build/PublishTargets.cs

```diff
@@ -309,21 +309,34 @@ public static BuildTargetResult PublishCoreHostPackages(BuildTargetContext c)
         {
             foreach (var file in Directory.GetFiles(Dirs.CorehostLocalPackages, "*.nupkg"))
             {
-                var hostBlob = $"{Channel}/Binaries/{SharedFrameworkNugetVersion}/{Path.GetFileName(file)}";
-                AzurePublisherTool.PublishFile(hostBlob, file);
-                Console.WriteLine($"Publishing package {hostBlob} to Azure.");
+                if (!IsRidAgnosticPackage(file) || EnvVars.PublishRidAgnosticPackages)
+                {
+                    var hostBlob = $"{Channel}/Binaries/{SharedFrameworkNugetVersion}/{Path.GetFileName(file)}";
+                    AzurePublisherTool.PublishFile(hostBlob, file);
+                    Console.WriteLine($"Publishing package {hostBlob} to Azure.");
+                }
+                else
+                {
+                    Console.WriteLine($"Skipping publishing package {file} to Azure because it is RID-agnostic and EnvVars.PublishRidAgnosticPackages is not set to true.");
+                }
             }
 
             return c.Success();
         }
 
+        private static bool IsRidAgnosticPackage(string packagePath)
+        {
+            var fileName = Path.GetFileName(packagePath);
+            return !fileName.StartsWith("runtime.", StringComparison.OrdinalIgnoreCase);
+        }
+
         [Target]
         public static BuildTargetResult PublishManagedPackages(BuildTargetContext c)
```

Comment:
No, this function only uploads the managed packages.

The nupkgs are split in our build output - everything that gets built from the `pkg` folder goes to `ROOT\artifacts\RID\corehost`.  While the packages that are built with `dotnet pack` go to `ROOT\artifacts\RID\packages`